include/deal.II/base/tensor.h

```diff
@@ -212,7 +212,12 @@ class Tensor<0,dim,Number>
   /**
    * Assignment from tensors with same underlying scalar type.
    */
-  Tensor &operator = (const Tensor<0,dim,Number> &rhs);
+  Tensor &operator = (const Tensor<0,dim,Number> &rhs)
+#ifndef __INTEL_COMPILER
+    = default;
+  // ICC 15 doesn't allow this opy constructor to be defaulted.
+  // see https://github.com/dealii/dealii/pull/5865.
+#endif
```

Comment:
Don't we need to add an
```
#else
;
#endif
```
otherwise we'd be missing the line terminator when `__INTEL_COMPILER` is defined?

Reply:
Perhaps since this function has a simple signature, may I suggest that we make this code block
```c++
#ifdef __INTEL_COMPILER
Tensor &operator = (const Tensor<0,dim,Number> &rhs);
#else
Tensor &operator = (const Tensor<0,dim,Number> &rhs) = default;
#endif
```
Then its a bit more clear what's going on, and the macro condition is the same here and later for the  definition.

Reply:
Fixed.